recipes/mp-units/2.2.0/conanfile.py

```diff
@@ -0,0 +1,267 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import default_cppstd, valid_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir
+
+required_conan_version = ">=1.59.0"
+
+
+def loose_lt_semver(v1, v2):
+    lv1 = [int(v) for v in v1.split(".")]
+    lv2 = [int(v) for v in v2.split(".")]
+    min_length = min(len(lv1), len(lv2))
+    return lv1[:min_length] < lv2[:min_length]
+
+
+class MPUnitsConan(ConanFile):
+    name = "mp-units"
+    homepage = "https://github.com/mpusz/mp-units"
+    description = "The quantities and units library for C++"
+    topics = (
+        "units",
+        "dimensions",
+        "quantities",
+        "dimensional-analysis",
+        "physical-quantities",
+        "physical-units",
+        "system-of-units",
+        "system-of-quantities",
+        "isq",
+        "si",
+        "library",
+        "quantity-manipulation",
+    )
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "cxx_modules": ["auto", True, False],
+        "std_format": ["auto", True, False],
+        "string_view_ret": ["auto", True, False],
+        "no_crtp": ["auto", True, False],
+        "contracts": ["none", "gsl-lite", "ms-gsl"],
+        "freestanding": [True, False],
+    }
+    default_options = {
+        "cxx_modules": "auto",
+        "std_format": "auto",
+        "string_view_ret": "auto",
+        "no_crtp": "auto",
```

Comment:
"auto" options are tricky in Conan Center, it's a pattern that perhaps should have not been used in other recipes either. However, this doesn't mean it's not supported, is more about the "auto' value should not be a value - would recommend the following, _on the condition_ that the real value (True/False) in these cases is derived based on the info from settings,

you can see an example here:
https://github.com/conan-io/conan-center-index/pull/16177/files

the behaviour will be the same , but basically:
* no need to have an "auto" value, `True`/`False` is fine - it reflects the actual options
* if one needs the default value to be set different depending on a setting, *do not* define a default option in the `default_options` dict
* in the `config_options()` method, give them a value `True`/`False` depending on the relevant `setting` (like compiler, compiler version)

This still behaves the same as "auto" - that is, "if the user doesnt explicitly give it a value, give it a default value based on.."

This will simplify the logic, as it would make the following redundant:
```    @property
    def _build_cxx_modules(self):
        return self._is_feature_enabled("cxx_modules")
```

now simply becomes checking for `self.options.cxx_modules` (it can only be true/False), and same for `_use_fmtlib`

I can also see that some of these options have a _default_ value (on/off) in the project's `CMakeLists.txt` - it may also be easier to just set the same values here, and avoid the auto logic - which is what we do for all other recipes where we can

Reply:
I am having issues in assigning a value to an option specified by the name:

```python
self.options.get_safe(name) = bool(
            min_version
            and Version(compiler.version) >= min_version
            and valid_min_cppstd(self, feature["min_cppstd"])
        )
```

The above does not work. How can I do that?

Reply:
You'll need to call the `setattr(self.options, name, newvalue)` built in method :)

Reply:
Thanks @RubenRBS, it seems I also have an issue on how to check if the option is not set in `config_options`. I checked things like `if self.options.get_safe(key) is None`, but it seems it is not the right way...

Reply:
Basically, I need something like this:

```python
        for key in self._option_feature_map.keys():
            if self.options.get_safe(key) is None:
                self._set_default_option(key)
```

Reply:
This is because the options do not have values set in the `config_options` method, you'll have to set it in the `configure()` one :)

Reply:
Ahh, right. I should not check here for `None`. Just set it up as a default, and then Conan will override it if user provided a value.

Reply:
Exactly! It should work as a programmatic "default_options", to give them a default value based on settings

Reply:
I am making those changes, but that also requires many changes in the project. So I will need to release 2.2.1 to make it work, which will take a while :-(